crates/turborepo-ui/src/tui/preferences.rs

```diff
@@ -0,0 +1,132 @@
+use std::{
+    fs::{self, File},
+    io::{BufReader, Write},
+};
+
+use serde::{Deserialize, Serialize};
+use serde_json::{from_reader, json, Value};
+use turbopath::AbsoluteSystemPathBuf;
+
+use super::task::TasksByStatus;
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct Preferences {
+    pub is_task_list_visible: Option<bool>,
+    pub active_task: Option<String>,
+    pub is_pinned_task_selection: Option<bool>,
+}
+
+#[derive(Debug)]
+pub enum PreferenceFields {
+    IsTaskListVisible,
+    ActiveTask,
+    PinnedTaskSelection,
+}
+
+impl Default for Preferences {
+    fn default() -> Self {
+        Self {
+            active_task: None,
+            is_task_list_visible: Some(true),
+            is_pinned_task_selection: Some(false),
+        }
+    }
+}
+
+const TUI_PREFERENCES_PATH_COMPONENTS: &[&str] = &[".turbo", "preferences", "tui.json"];
+
+fn read_json(path: &AbsoluteSystemPathBuf) -> Preferences {
+    File::open(path)
+        .ok()
+        .and_then(|file| from_reader(BufReader::new(file)).ok())
+        .unwrap_or_default()
+}
+
+impl Preferences {
+    pub fn update_preference(
+        repo_root: &AbsoluteSystemPathBuf,
+        field: PreferenceFields,
+        new_value: Value,
+    ) -> Result<(), Box<dyn std::error::Error>> {
+        // TODO: Clean these up, should be taken from constants
+        let preferences_dir = repo_root.join_components(&[".turbo", "preferences"]);
+        let preferences_file = repo_root.join_components(&[".turbo", "preferences", "tui.json"]);
+
+        fs::create_dir_all(preferences_dir.as_std_path())?;
```

Comment:
Generally we should prefer methods on the `turbopath`s types since we have nicer error messages. (By default Rust io errors don't include paths in them which can lead to frustrating error messages).

```suggestion
        preferences_file.ensure_dir()?;
```